api/v1alpha4/openstackcluster_webhook.go

```diff
@@ -39,8 +39,8 @@ func (r *OpenStackCluster) SetupWebhookWithManager(mgr manager.Manager) error {
 		Complete()
 }
 
-// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha4-openstackcluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=openstackclusters,versions=v1alpha4,name=validation.openstackcluster.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1beta1
-// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1alpha4-openstackcluster,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=openstackclusters,versions=v1alpha4,name=default.openstackcluster.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1beta1
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha4-openstackcluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=openstackclusters,versions=v1alpha4,name=v1alpha4.openstackcluster.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1beta1
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1alpha4-openstackcluster,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=openstackclusters,versions=v1alpha4,name=v1alpha4.openstackcluster.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1beta1
```

Comment:
does this means we have to do this for a1alpha5 again  when we have a1alpha6 out?

Reply:
Hi @jichenjc 

i just take a look how CAPA is doing it. They only have one webhooks for the current apiVersion. For older apiVersions they don't have any webhook anymore:

Mainbranch - v1beta1: https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/api/v1beta1
Mainbranch - v1alpha4: https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/api/v1alpha4

Commit: https://github.com/kubernetes-sigs/cluster-api-provider-aws/commit/4274a5ab5c9dc391045840d6e45e17fe0cbab3dc?diff=split

Reply:
Thanks for sharing, seems we can follow their pattern in our code as well..